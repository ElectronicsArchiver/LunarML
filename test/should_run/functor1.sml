functor Double (S : sig
                    type t
                    val f : t -> t
                end) = struct
val ff = fn x => S.f (S.f x)
end
structure F = Double (struct
                       type t = int
                       fun f x = x + x
                       end)
structure G = Double (type t = int
                      fun f x = x * x
                     )
structure H = Double (type t = string
                      fun f x = x ^ x
                     )
val _ = print (Int.toString (F.ff 42) ^ "\n");
val _ = print (Int.toString (G.ff 4) ^ "\n");
val _ = print (H.ff "x" ^ "\n");
