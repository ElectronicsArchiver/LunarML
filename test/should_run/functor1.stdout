168
256
xxxx
